src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -818,7 +818,7 @@ internal MethodSymbol AsMember(NamedTypeSymbol newOwner)
         {
             Debug.Assert(this.IsDefinition);
             Debug.Assert(ReferenceEquals(newOwner.OriginalDefinition, this.ContainingSymbol.OriginalDefinition));
-            return (newOwner == this.ContainingSymbol) ? this : new SubstitutedMethodSymbol((SubstitutedNamedTypeSymbol)newOwner, this);
+            return (newOwner == this.ContainingSymbol) ? this : new SubstitutedMethodSymbol(newOwner, this);
```

Comment:
Why would you say that? With the unnecessarily tight coupling fixed, the behavior appears to be precisely as needed in the callers. I suspect you're imagining some value in pushing some additional error recovery code into the callers (which would be duplicate code in each caller), but you have not identified that value.

The questions are
1. Is the bug fixed? (If not, please demonstrate)
2. Is the fix tested?